contracts/smart-account/test/upgrades/v1/SmartAccountV1.sol

```diff
@@ -116,17 +100,39 @@ contract SmartAccountV1 is
     }
 
     /**
-     * @dev This function allows the owner or entry point to execute certain actions.
-     * If the caller is not authorized, the function will revert with an error message.
-     * @notice This modifier is marked as internal and can only be called within the contract itself.
+     * @dev Constructor that sets the owner of the contract and the entry point contract.
+     * @param anEntryPoint The address of the entry point contract.
      */
-    function _requireFromEntryPointOrOwner() internal view {
-        if (msg.sender != address(entryPoint()) && msg.sender != owner)
-            revert CallerIsNotEntryPointOrOwner(msg.sender);
+    constructor(IEntryPoint anEntryPoint) {
+        _self = address(this);
+        // By setting the owner it is not possible to call init anymore,
+        // so we create an account with fixed non-zero owner.
+        // This is an unusable account, perfect for the singleton
+        owner = address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+        if (address(anEntryPoint) == address(0))
+            revert EntryPointCannotBeZero();
+        _entryPoint = anEntryPoint;
+        _chainId = block.chainid;
+    }
+
+    /// Getters
+    /**
+     * @dev Returns the address of the implementation contract associated with this contract.
+     * @notice The implementation address is stored in the contract's storage slot with index 0.
+     */
+    function getImplementation()
+        external
+        view
+        returns (address _implementation)
+    {
+        // solhint-disable-next-line no-inline-assembly
```

Comment:
can be removed as well

Reply:
?

Reply:
the solhint rule:        // solhint-disable-next-line no-inline-assembly

Reply:
got it

Reply:
so remove it in entire codebase?